visualizer/src/Controls/Tracking/LabelTimeline.js

```diff
@@ -0,0 +1,65 @@
+import { Box } from '@material-ui/core';
+import { useSelector } from '@xstate/react';
+import React from 'react';
+import { useDivision, useFeature, useImage, useLabeled } from '../../ProjectContext';
+
+function useColors() {
+  const labeled = useLabeled();
+  const featureIndex = useSelector(labeled, (state) => state.context.feature);
+  const feature = useFeature(featureIndex);
+  const colors = useSelector(feature, (state) => state.context.colors);
+  return colors;
+}
+
+function FrameBox({ frame, numFrames, color }) {
+  const image = useImage();
+
+  const boxStyle = {
+    position: 'absolute',
+    backgroundColor: color,
+    height: '10px',
+    width: `${(1 / numFrames) * 100}%`,
+    left: `${(frame / numFrames) * 100}%`,
+    pointerEvents: 'none',
+  };
+
+  const onClick = (event, newValue) => {
+    if (newValue !== frame) {
+      image.send({ type: 'LOAD_FRAME', frame: newValue });
+    }
+  };
+
+  return <Box style={boxStyle} onClick={onClick}></Box>;
+}
+
+function LabelTimeline({ label }) {
+  const image = useImage();
+  const numFrames = useSelector(image, (state) => state.context.numFrames);
+
+  const colors = useColors();
+  const division = useDivision(label);
+  const { frames } = division;
+  const color = colors[label] ?? '#000000';
+
+  return (
+    <Box
+      style={{
+        position: 'relative',
+        display: 'flex',
+        height: '10px',
+        width: '100%',
+      }}
+    >
+      {[...Array(numFrames).keys()].map((frame) => (
+        <FrameBox
+          frame={frame}
+          numFrames={numFrames}
+          color={frames.includes(frame) ? color : '#FFFFFF'}
+          key={frame}
+        />
+      ))}
+    </Box>
+  );
+}
+
+export default React.memo(LabelTimeline);
```

Comment:
Can you comment on why this is a `memo`?

Reply:
The divisions UI tends to rerender too often. For example, when moving the mouse over a different label on the canvas, the React dev tools show these components rerendering because some of their hooks changed, but no data or state relevant to the division changed, so ideally the components should not rerender. While developing, I was hoping that memo could help out with this issue. After revisiting it, it seems that the component still rerenders even though it is memoized...

The React component profiler shows that the rerenders take about 10 milliseconds, which should not be noticeable as it's less than the 16 ms between frames, but it could contribute to jank if are other components rerendering at the same time. I'll post an issue to keep track of this so we can fix this in a future release.

Reply:
Added issues #293 to track it outside of this PR